src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/UpdateRepresentationService.java

```diff
@@ -0,0 +1,159 @@
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.APPLICANT_REPRESENTED;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.APP_SOLICITOR_POLICY;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.CONSENTED_RESPONDENT_REPRESENTED;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.CONSENTED_SOLICITOR_ADDRESS;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.CONSENTED_SOLICITOR_NAME;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.CONTESTED_RESPONDENT_REPRESENTED;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.CONTESTED_SOLICITOR_ADDRESS;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.CONTESTED_SOLICITOR_EMAIL;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.CONTESTED_SOLICITOR_NAME;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.RESP_SOLICITOR_ADDRESS;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.RESP_SOLICITOR_EMAIL;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.RESP_SOLICITOR_NAME;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.SOLICITOR_EMAIL;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class UpdateRepresentationService {
+
+    private final AuditEventService auditEventService;
+    private final IdamAuthService idamClient;
+    private final ObjectMapper objectMapper;
+    private final CaseDataService caseDataService;
+    private final PrdOrganisationService organisationService;
+    private final UpdateSolicitorDetailsService updateSolicitorDetailsService;
+    private final ChangeOfRepresentationService changeOfRepresentationService;
+
+    private static final String CHANGE_REQUEST_FIELD = "changeOrganisationRequestField";
+    private static final String NOC_EVENT = "nocRequest";
+    private static final String CHANGE_OF_REPS = "ChangeOfRepresentatives";
+
+    private boolean isApplicant;
+
+    public Map<String, Object> updateRepresentationAsSolicitor(CaseDetails caseDetails,
+                                                               String authToken)  {
+
+        log.info("Updating representation for case ID {}", caseDetails.getId());
+
+        AuditEvent auditEvent = auditEventService.getLatestAuditEventByName(caseDetails.getId().toString(), NOC_EVENT)
+            .orElseThrow(() -> new IllegalStateException(String.format("Could not find %s event in audit", NOC_EVENT)));
+
+        UserDetails solicitorToAdd = idamClient.getUserByUserId(authToken, auditEvent.getUserId());
+        ChangeOrganisationRequest change = getChangeOrganisationRequest(caseDetails);
```

Comment:
pettyComment: would probably call this variable changeRequest, change is a bit vague